src/server/mpsserver.ts

```diff
@@ -177,7 +184,9 @@ export class MPSServer {
   onClose = async (socket: CIRASocket): Promise<void> => {
     logger.debug('MPS:CIRA connection closed')
     try {
-      await this.handleDeviceDisconnect(socket.tag.nodeid)
+      if (devices[socket.tag.nodeid]?.ciraSocket?.tag.id === socket?.tag.id) {
```

Comment:
Do you think we should log on the else condition to know when this is happening?

Reply:
Yes, I think it will be good to have that. Should I do a separate PR for adding/modifying logs around CIRA connections (especially around all socket events)?

Reply:
I would do it as part of this one.

Reply:
I've added extra logging around cira connection events with the new commit, let me know your feedback on the changes